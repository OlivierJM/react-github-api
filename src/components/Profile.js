// this file will be a holder for user's profile info as indicated in the issues
