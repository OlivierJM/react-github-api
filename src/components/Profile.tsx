// this file will be a holder for user's profile info as indicated in the issues
import * as React from "react"

interface profileProps {
  name: string
  orgs: number
  location: string
  totalCount: number
}

const Profile: React.SFC<profileProps> = props => (
  <div>
    <h2>{props.name}</h2>
    <p>
      Organizations: {props.orgs} <br />
      Contributions: {props.location} <br />
      Number of Repos: {props.totalCount}
    </p>
    {/* <img src={props.image} alt="github avatar" /> */}
  </div>
)

export default Profile
