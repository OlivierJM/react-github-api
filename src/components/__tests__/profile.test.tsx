import * as React from "react"
import { shallow, mount } from "enzyme"
import Profile from "../Profile"

describe("Profile component", () => {
  const profile = mount(
    <Profile name="olivier" orgs={1} location="lsk" totalCount={10} />
  )
  it("should match the snapshot", () => {
    expect(profile).toMatchSnapshot()
  })
  it("should render proper props", () => {
    expect(profile.prop("orgs")).toEqual(1)
  })
  it("should contain the profile classs", () => {
    expect(profile.find(".profile").exists()).toBe(true)
  })
})
// expect(nav.findWhere(el => el.is(NavItem) && el.prop('url') === '/login').length).toEqual(1)
