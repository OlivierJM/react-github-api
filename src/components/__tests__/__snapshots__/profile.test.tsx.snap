// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should match the snapshot 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Profile
    location="lsk"
    name="olivier"
    orgs={1}
    totalCount={10}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "location": "lsk",
      "name": "olivier",
      "orgs": 1,
      "totalCount": 10,
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="profile"
      >
        <h2>
          olivier
        </h2>
        <p>
          Organizations: 
          1
           
          <br />
          Contributions: 
          lsk
           
          <br />
          Number of Repos: 
          10
        </p>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            olivier
          </h2>,
          <p>
            Organizations: 
            1
             
            <br />
            Contributions: 
            lsk
             
            <br />
            Number of Repos: 
            10
          </p>,
        ],
        "className": "profile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <h2>
            olivier
          </h2>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "olivier",
          },
          "ref": null,
          "rendered": Array [
            "olivier",
          ],
          "type": "h2",
        },
        Object {
          "instance": <p>
            Organizations: 
            1
             
            <br />
            Contributions: 
            lsk
             
            <br />
            Number of Repos: 
            10
          </p>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Organizations: ",
              1,
              " ",
              <br />,
              "Contributions: ",
              "lsk",
              " ",
              <br />,
              "Number of Repos: ",
              10,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Organizations: ",
            "1",
            " ",
            Object {
              "instance": <br />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "br",
            },
            "Contributions: ",
            "lsk",
            " ",
            Object {
              "instance": <br />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "br",
            },
            "Number of Repos: ",
            "10",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "location": "lsk",
        "name": "olivier",
        "orgs": 1,
        "totalCount": 10,
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="profile"
        >
          <h2>
            olivier
          </h2>
          <p>
            Organizations: 
            1
             
            <br />
            Contributions: 
            lsk
             
            <br />
            Number of Repos: 
            10
          </p>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              olivier
            </h2>,
            <p>
              Organizations: 
              1
               
              <br />
              Contributions: 
              lsk
               
              <br />
              Number of Repos: 
              10
            </p>,
          ],
          "className": "profile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <h2>
              olivier
            </h2>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "olivier",
            },
            "ref": null,
            "rendered": Array [
              "olivier",
            ],
            "type": "h2",
          },
          Object {
            "instance": <p>
              Organizations: 
              1
               
              <br />
              Contributions: 
              lsk
               
              <br />
              Number of Repos: 
              10
            </p>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Organizations: ",
                1,
                " ",
                <br />,
                "Contributions: ",
                "lsk",
                " ",
                <br />,
                "Number of Repos: ",
                10,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Organizations: ",
              "1",
              " ",
              Object {
                "instance": <br />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "br",
              },
              "Contributions: ",
              "lsk",
              " ",
              Object {
                "instance": <br />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "br",
              },
              "Number of Repos: ",
              "10",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
